Rebus.LegacyExtensions/RabbitMqEnhacedSubscriptionStorage.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+using HermaFx;
+
+using Rebus.Logging;
+using Rebus.Subscriptions;
+using Rebus.Transport;
+
+namespace Rebus
+{
+	// Modified Subscription storage for subscribing/unsubscribing in OneExchangePerClass model
+	public class RabbitMqEnhacedSubscriptionStorage : ISubscriptionStorage, IDisposable
+	{
+		private readonly ISubscriptionStorage _inner;
+		private readonly ILog _log;
+		private readonly ITransport _transport;
+		private readonly RabbitMqMangler _mangler;
+
+		#region Reflection fields
+
+		private FieldInfo DirectExchangeField = AppDomain.CurrentDomain.GetAssemblies()
```

Comment:
I do not understand this.. for OneExchangePerClass, we should be binding exchanges named after each message to our own Inbound exchange. So, why do you need _directExchangeName at all here?